Modules/posixmodule.c

```diff
@@ -1927,11 +1927,13 @@ _pystat_fromstructstat(STRUCT_STAT *st)
         return NULL;
 
     PyStructSequence_SET_ITEM(v, 0, PyLong_FromLong((long)st->st_mode));
-#ifdef HAVE_LARGEFILE_SUPPORT
+#if defined(HAVE_LARGEFILE_SUPPORT) || defined(MS_WINDOWS)
+    assert(sizeof(st->st_ino) >= sizeof(unsigned long long));
```

Comment:
Use `Py_BUILD_ASSERT`.

Should test `sizeof(st->st_ino) <= sizeof(unsigned long long)`.